charon/src/ast/llbc_ast_utils.rs

```diff
@@ -38,29 +38,38 @@ pub fn combine_switch_targets_span(targets: &Switch) -> Span {
 }
 
 impl Switch {
-    pub fn get_targets(&self) -> Vec<&Statement> {
+    pub fn iter_targets(&self) -> impl Iterator<Item = &Statement> {
+        use itertools::Either;
         match self {
-            Switch::If(_, exp1, exp2) => {
-                vec![exp1, exp2]
-            }
-            Switch::SwitchInt(_, _, targets, otherwise) => {
-                let mut out: Vec<&Statement> = vec![];
-                for (_, tgt) in targets {
-                    out.push(tgt);
-                }
-                out.push(otherwise);
-                out
-            }
-            Switch::Match(_, targets, otherwise) => {
-                let mut out: Vec<&Statement> = vec![];
-                for (_, tgt) in targets {
-                    out.push(tgt);
-                }
-                if let Some(otherwise) = otherwise {
-                    out.push(otherwise);
-                }
-                out
-            }
+            Switch::If(_, exp1, exp2) => Either::Left([&**exp1, &**exp2].into_iter()),
+            Switch::SwitchInt(_, _, targets, otherwise) => Either::Right(Either::Left(
+                targets.iter().map(|(_, tgt)| tgt).chain([&**otherwise]),
+            )),
+            Switch::Match(_, targets, otherwise) => Either::Right(Either::Right(
+                targets
+                    .iter()
+                    .map(|(_, tgt)| tgt)
+                    .chain(otherwise.as_deref()),
+            )),
```

Comment:
Why do you use `Either` in this code?

Reply:
Because each branch returns a different iterator type, so if I didn't use `Either` I'd get a type error. `Either` merges two possible iterator types into one

Reply:
Indeed, thanks!